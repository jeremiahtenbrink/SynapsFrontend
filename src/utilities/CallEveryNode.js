/**
 * callEveryNode
 *
 * Calls the callback for every el and child el in the element. Including
 * the first el passed in.
 * @property {HTMLElement} el
 * @property {Function} cb
 */
export const callEveryNode = ( el, cb ) => {
  cb( el );
  if( el.hasChildNodes() ){
    el.childNodes.forEach( child => {
      callEveryNode( child, cb );
    } );
  }
};