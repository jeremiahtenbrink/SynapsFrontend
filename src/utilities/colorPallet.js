/**
 * @typedef {Object} ColorPallet
 * @property {Color} PRIMARY_COLOR_DARKER_10
 * @property {Color} PRIMARY_COLOR_DARKER_9
 * @property {Color} PRIMARY_COLOR_DARKER_8
 * @property {Color} PRIMARY_COLOR_DARKER_7
 * @property {Color} PRIMARY_COLOR_DARKER_6
 * @property {Color} PRIMARY_COLOR_DARKER_5
 * @property {Color} PRIMARY_COLOR_DARKER_4
 * @property {Color} PRIMARY_COLOR_DARKER_3
 * @property {Color} PRIMARY_COLOR_DARKER_2
 * @property {Color} PRIMARY_COLOR_DARKER_1
 * @property {Color} PRIMARY_COLOR
 * @property {Color} PRIMARY_COLOR_LIGHTER_1
 * @property {Color} PRIMARY_COLOR_LIGHTER_2
 * @property {Color} PRIMARY_COLOR_LIGHTER_3
 * @property {Color} PRIMARY_COLOR_LIGHTER_4
 * @property {Color} PRIMARY_COLOR_LIGHTER_5
 * @property {Color} PRIMARY_COLOR_LIGHTER_6
 * @property {Color} PRIMARY_COLOR_LIGHTER_7
 * @property {Color} PRIMARY_COLOR_LIGHTER_8
 * @property {Color} PRIMARY_COLOR_LIGHTER_9
 * @property {Color} PRIMARY_COLOR_LIGHTER_10
 * @property {Color} SECONDARY_COLOR_DARKER_10
 * @property {Color} SECONDARY_COLOR_DARKER_9
 * @property {Color} SECONDARY_COLOR_DARKER_8
 * @property {Color} SECONDARY_COLOR_DARKER_7
 * @property {Color} SECONDARY_COLOR_DARKER_6
 * @property {Color} SECONDARY_COLOR_DARKER_5
 * @property {Color} SECONDARY_COLOR_DARKER_4
 * @property {Color} SECONDARY_COLOR_DARKER_3
 * @property {Color} SECONDARY_COLOR_DARKER_2
 * @property {Color} SECONDARY_COLOR_DARKER_1
 * @property {Color} SECONDARY_COLOR
 * @property {Color} SECONDARY_COLOR_LIGHTER_1
 * @property {Color} SECONDARY_COLOR_LIGHTER_2
 * @property {Color} SECONDARY_COLOR_LIGHTER_3
 * @property {Color} SECONDARY_COLOR_LIGHTER_4
 * @property {Color} SECONDARY_COLOR_LIGHTER_5
 * @property {Color} SECONDARY_COLOR_LIGHTER_6
 * @property {Color} SECONDARY_COLOR_LIGHTER_7
 * @property {Color} SECONDARY_COLOR_LIGHTER_8
 * @property {Color} SECONDARY_COLOR_LIGHTER_9
 * @property {Color} SECONDARY_COLOR_LIGHTER_10
 * @property {Color} WHITE_DARKER_10
 * @property {Color} WHITE_DARKER_9
 * @property {Color} WHITE_DARKER_8
 * @property {Color} WHITE_DARKER_7
 * @property {Color} WHITE_DARKER_6
 * @property {Color} WHITE_DARKER_5
 * @property {Color} WHITE_DARKER_4
 * @property {Color} WHITE_DARKER_3
 * @property {Color} WHITE_DARKER_2
 * @property {Color} WHITE_DARKER_1
 * @property {Color} WHITE
 * @property {GreenToBlueGradient} GREEN_TO_BLUE_GRADIENT
 */
export const colorPallet = {
  PRIMARY_COLOR_DARKER_10: "#10131c",
  PRIMARY_COLOR_DARKER_9: "#141822",
  PRIMARY_COLOR_DARKER_8: "#181c29",
  PRIMARY_COLOR_DARKER_7: "#1c212f",
  PRIMARY_COLOR_DARKER_6: "#1f2535",
  PRIMARY_COLOR_DARKER_5: "#232a3c",
  PRIMARY_COLOR_DARKER_4: "#272e42",
  PRIMARY_COLOR_DARKER_3: "#2b3349",
  PRIMARY_COLOR_DARKER_2: "#2e374f",
  PRIMARY_COLOR_DARKER_1: "#323c56",
  PRIMARY_COLOR: "#36405C",
  PRIMARY_COLOR_LIGHTER_1: "#3a4462",
  PRIMARY_COLOR_LIGHTER_2: "#3e4969",
  PRIMARY_COLOR_LIGHTER_3: "#414d6f",
  PRIMARY_COLOR_LIGHTER_4: "#455276",
  PRIMARY_COLOR_LIGHTER_5: "#49567c",
  PRIMARY_COLOR_LIGHTER_6: "#4d5b83",
  PRIMARY_COLOR_LIGHTER_7: "#505f89",
  PRIMARY_COLOR_LIGHTER_8: "#54648f",
  PRIMARY_COLOR_LIGHTER_9: "#586896",
  PRIMARY_COLOR_LIGHTER_10: "#5c6d9c",
  
  SECONDARY_COLOR_DARKER_10: "#2d6f61",
  SECONDARY_COLOR_DARKER_9: "#307667",
  SECONDARY_COLOR_DARKER_8: "#337d6d",
  SECONDARY_COLOR_DARKER_7: "#368574",
  SECONDARY_COLOR_DARKER_6: "#398c7a",
  SECONDARY_COLOR_DARKER_5: "#3c9380",
  SECONDARY_COLOR_DARKER_4: "#3f9a86",
  SECONDARY_COLOR_DARKER_3: "#42a28d",
  SECONDARY_COLOR_DARKER_2: "#45a993",
  SECONDARY_COLOR_DARKER_1: "#48b099",
  SECONDARY_COLOR: "#4CB69F",
  SECONDARY_COLOR_LIGHTER_1: "#53b9a3",
  SECONDARY_COLOR_LIGHTER_2: "#5abca7",
  SECONDARY_COLOR_LIGHTER_3: "#62bfab",
  SECONDARY_COLOR_LIGHTER_4: "#69c2af",
  SECONDARY_COLOR_LIGHTER_5: "#70c5b2",
  SECONDARY_COLOR_LIGHTER_6: "#77c8b6",
  SECONDARY_COLOR_LIGHTER_7: "#7fcbba",
  SECONDARY_COLOR_LIGHTER_8: "#86cebe",
  SECONDARY_COLOR_LIGHTER_9: "#8dd1c2",
  SECONDARY_COLOR_LIGHTER_10: "#94d4c6",
  
  WHITE_DARKER_10: "#cac4b9",
  WHITE_DARKER_9: "#cfc9bf",
  WHITE_DARKER_8: "#d3cec4",
  WHITE_DARKER_7: "#d7d3ca",
  WHITE_DARKER_6: "#dcd8d0",
  WHITE_DARKER_5: "#e0ddd6",
  WHITE_DARKER_4: "#e5e2dc",
  WHITE_DARKER_3: "#e9e6e2",
  WHITE_DARKER_2: "#edebe7",
  WHITE_DARKER_1: "#f2f0ed",
  WHITE: "#F6F5F3",
  
  /**
   * @typedef {Object} GreenToBlueGradient
   * @property {object} Colors
   * @property {string} backgroundColor
   * @property {string} backgroundImage
   */
  GREEN_TO_BLUE_GRADIENT: {
    
    /**
     * @typedef {Object} Colors
     * @property {Color} color_1
     * @property {Color} color_2
     */
    colors: {
      color_1: "#00EFA9", color_2: "#3F56F0",
    },
    backgroundColor: "#00EFA9",
    backgroundImage: "linear-gradient(bottom right, #00EFA9, #3F56F0)",
  },
};