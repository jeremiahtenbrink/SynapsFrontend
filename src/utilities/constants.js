/**
 * @typedef {number} LogType
 *
 */

import { css } from "styled-components";
import { ButtonTheme } from "../components";
import theme from "styled-theming";
import { mapToTheme as map } from "styled-map";

/**
 * @typedef {object} LOG_TYPES
 * @param {logType} INFO
 * @param {logType} WARNING
 * @param {logType} ERROR
 * @param {logType} OBJECT
 */
export const LOG_TYPES = {
  VERBOSE: 1, INFO: 2, WARNING: 4, ERROR: 5, OBJECT: 6, PLAIN: 3,
};

/**
 * @typedef {string} DEBUG_LOG_SIZE
 *
 */

/**
 * @typedef {object} DEBUG_LOG_SIZES
 * @param {DEBUG_LOG_SIZE} LARGE
 * @param {DEBUG_LOG_SIZE} MEDIUM
 * @param {DEBUG_LOG_SIZE} NORMAL
 * @param {DEBUG_LOG_SIZE} SMALL
 */
export const DEBUG_LOG_SIZES = {
  LARGE: "large", MEDIUM: "medium", NORMAL: "normal", SMALL: "small",
};

/**
 * @typedef {("APP_VIEW_DESKTOP" | "APP_VIEW_MOBILE")} AppView
 *
 */

/**
 * @type {{NAV_BAR_HEIGHT: number, MAX_DASHBOARD_CONTAINER_WIDTH: number,
 *   NAV_BAR_DARK: string, FOOTER_HEIGHT: number, BRAIN_PIC_DARK: string,
 *   NAV_BAR_LIGHT: string, BRAIN_PIC_LIGHT: string, PRIMARY_COLOR: string,
 *   SYNAPS_DARK: string, SECONDARY_DARKER1: string, SECONDARY_COLOR: string,
 *   SYNAPS_LIGHT: string}}
 */
let THEMEprep = {
  
  secondaryButtonBg: "#4CB69F",
  
  light: "#4CB69F", dark: "#197d67",
  
  CONTAINER_BG: "white",
  
  PRIMARY_COLOR: "#0d2545", PRIMARY_COLOR_LIGHTER1: "#36405c",
  
  TEXT_LIGHT: "#F6F5F3", TEXT_DARK: "#323C56", TEXT_SECONDARY_COLOR: "#4CB69F",
  
  COLOR_WHITE: "#f0ede6",
  
  SECONDARY_COLOR: "#A2D8C7", SECONDARY_DARKER1: "#4CB69F",
  
  NAV_BAR_HEIGHT: 75, FOOTER_HEIGHT: 30,
  
  NAV_BAR_DARK: "#0C2545", NAV_BAR_LIGHT: "#F6F5F3",
  
  SYNAPS_DARK: "#36405C", SYNAPS_LIGHT: "#FFFFFF",
  
  BRAIN_PIC_DARK: "#164167", BRAIN_PIC_LIGHT: "#e1ded7",
  
  MAX_DASHBOARD_CONTAINER_WIDTH: 1140,
};

THEMEprep = {
  ...THEMEprep, flexCenterColumn: css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`,
};

export const THEME = { ...THEMEprep, ...ButtonTheme };

/**
 * @typedef {Object} ThemeState
 * @property {Color} PRIMARY_COLOR
 * @property {Color} primaryColorB98C4
 * @property {Color} primaryColor86869A
 * @property {Color} primaryColor5C5F78
 * @property {Color} primaryColor36405C
 * @property {Color} primaryColor293046
 * @property {Color} primaryColor353544
 *
 * @property {Color} SECONDARY_COLOR
 * @property {Color} secondaryColorEAF5F1
 * @property {Color} secondaryColorD8EEE6
 * @property {Color} secondaryColorC6E6DB
 * @property {Color} secondaryColorB5DFD1
 * @property {Color} secondaryColor92B2AA
 * @property {Color} secondaryColor798A87
 *
 * @property {Color} white
 * @property {Color} grayF1F2F2
 * @property {Color} grayE6E7E8
 * @property {Color} grayD1D3D4
 * @property {Color} grayBCBEC0
 * @property {Color} grayA7A9AC
 * @property {Color} gray939598
 *
 * @property {number} largeRadius
 * @property {number} smallRadius
 * @property {number} NAV_BAR_HEIGHT
 * @property {number} FOOTER_HEIGHT
 *
 * @property {string} NAV_BAR_LIGHT
 * @property {string} NAV_BAR_DARK
 *
 * @property {THEMING_VALUES} NAV_STYLE
 *
 * @property {Color} BRAIN_PIC_DARK
 * @property {Color} BRAIN_PIC_LIGHT
 *
 */

export const APP_VIEW_MOBILE = "APP_VIEW_MOBILE";
export const APP_VIEW_DESKTOP = "APP_VIEW_DESKTOP";

/**
 * @typedef {string} APP_PATH
 */

/**
 * @typedef {object.<{APP_PATH}, string>} AppPaths
 */
export const APP_PATHS = {
  SIGN_UP_PATH: "/signup",
  SIGN_IN_PATH: "/signin",
  DASHBOARD_PATH: "/dashboard",
  CREATE_DECK_PATH: "/create/deck",
  PREVIEW_DECK_PATH: "/preview",
  TESTING: "/test",
  LANDING_PAGE: "/",
  QUIZ_MODE: "/quiz-mode",
};

/**
 * @type Sizes
 *
 */
export const SIZES = {
  mobileS: 320,
  mobileM: 375,
  mobileL: 425,
  tablet: 768,
  laptop: 1024,
  laptopL: 1440,
  desktop: 2560,
};
/**
 * @category Utilities
 * @type {Devices}
 */
export const MEDIA_QUERIES = {
  mobileS: `(min-width: ${ SIZES.mobileS }px)`,
  mobileM: `(min-width: ${ SIZES.mobileM }px)`,
  mobileL: `(min-width: ${ SIZES.mobileL }px)`,
  tablet: `(min-width: ${ SIZES.tablet }px)`,
  laptop: `(min-width: ${ SIZES.laptop }px)`,
  laptopL: `(min-width: ${ SIZES.laptopL }px)`,
  desktop: `(min-width: ${ SIZES.desktop }px)`,
};

/**
 * @typedef {function} Dispatch
 */

/**
 * @typedef {string} Color
 */

/**
 * @typedef {object} Sizes
 * @property {number} mobileS '320px',
 * @property {number} mobileM '375px',
 * @property {number} mobileL '425px',
 * @property {number} tablet '768px',
 * @property {number} laptop '1024px',
 * @property {number} laptopL '1440px',
 * @property {number} desktop '2560px',
 */

/**
 * @typedef {string} MediaQuery
 */

/**
 * @typedef {object} Devices
 * @property {MediaQuery} mobileS   320px
 * @property {MediaQuery} mobileM   375px
 * @property {MediaQuery} mobileL   425px
 * @property {MediaQuery} tablet    768px
 * @property {MediaQuery} laptop    1024px
 * @property {MediaQuery} laptopL   1440px
 * @property {MediaQuery} desktop   2560px
 */
