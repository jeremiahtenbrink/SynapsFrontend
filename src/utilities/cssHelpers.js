import { css } from "styled-components";

export const flex = ( { flexDirection, justifyContent, alignItmes, flexWrap } ) => {
  let rtrString = css`
display: flex;
flex-direction: ${ flexDirection };
`;
  if( justifyContent ){
    rtrString += css`
justify-content: ${ justifyContent };
`;
  }
  
  if( alignItmes ){
    rtrString += css`
align-items: ${ alignItmes };
`;
  }
  if( flexWrap ){
    rtrString += css`
flex-wrap: ${ flexWrap };
`;
  }
  return rtrString;
};