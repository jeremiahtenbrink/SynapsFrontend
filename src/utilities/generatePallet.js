const polished = require( "polished" );
const fs = require( "fs" );
const { pascalCase } = require( "change-case" );
const { darken, lighten, linearGradient } = polished;

/**
 * @param {Object} colors
 */
const generateColorPallet = ( colors ) => {
  
  const stringArray = generateObjectStartAndClose( "colorPallet",
    ( name, fileStrArr, docStringArray, tabs ) => {
      Object.keys( colors ).forEach( key => {
        
        generateDarker( key,
          colors[ key ],
          fileStrArr,
          docStringArray,
          tabs,
          10,
          .02
        );
        
        addPropertyDocString( key, "Color", docStringArray, tabs - 1 );
        addKeyValueToFileStr( key, colors[ key ], fileStrArr, tabs );
        
        generateLighter( key,
          colors[ key ],
          fileStrArr,
          docStringArray,
          tabs,
          10,
          .02
        );
        
        addStrToArrayStr( "", fileStrArr, tabs );
      } );
      
      generateDarker( "WHITE",
        "#F6F5F3",
        fileStrArr,
        docStringArray,
        tabs,
        10,
        .02
      );
      
      addPropertyDocString( "WHITE", "Color", docStringArray, tabs - 1 );
      addKeyValueToFileStr( "WHITE", "#F6F5F3", fileStrArr, tabs );
      
      addPropertyDocString( "GREEN_TO_BLUE_GRADIENT",
        pascalCase( "GREEN_TO_BLUE_GRADIENT" ),
        docStringArray,
        tabs - 1
      );
      addGradientToFileStr( "GREEN_TO_BLUE_GRADIENT", {
        colorStops: [ "#00EFA9", "#3F56F0" ],
        fallback: "#00EFA9",
        toDirection: "bottom right"
      }, fileStrArr, docStringArray, tabs );
      
    },
    0
  );
  
  fs.writeFileSync( "colorPallet.js", stringArray.join( "\n" ) );
  
};

/**
 * @typedef {Object} GradientConfig
 * @property {string []} colorStops
 * @property {string} toDirection
 * @property {string } fallback
 *
 */

/**
 *
 * @param {string} name
 * @param {GradientConfig} config
 * @param {string []} fileStrArr
 * @param {string []} docStringArray
 * @param {number} tabs
 */
const addGradientToFileStr = ( name, config, fileStrArr, docStringArray,
  tabs = 0 ) => {
  let gradient = linearGradient( config );
  
  const gradientString = generateObjectStartAndClose( name,
    ( gradientName, gradientStringArray, gradientDocStringArr,
      gradientTabs ) => {
      
      const colorString = generateObjectStartAndClose( "colors",
        ( colorsName, colorStringArr, colorDocStringArr, colorTabs ) => {
          
          config.colorStops.forEach( ( color, i ) => {
            const newKey = `color_${ i + 1 }`;
            addPropertyDocString( newKey,
              "Color",
              colorDocStringArr,
              colorTabs - 1
            );
            addKeyValueToFileStr( newKey,
              config.colorStops[ i ],
              colorStringArr,
              colorTabs
            );
          } );
        },
        gradientTabs,
        true,
        name
      );
      addPropertyDocString( pascalCase( "colors" ),
        "object",
        gradientDocStringArr,
        gradientTabs - 1
      );
      gradientStringArray.push( " " );
      colorString.forEach( item => {
        gradientStringArray.push( item );
      } );
      addPropertyDocString( "backgroundColor",
        "string",
        gradientDocStringArr,
        gradientTabs - 1
      );
      addKeyValueToFileStr( "backgroundColor",
        gradient[ "backgroundColor" ],
        gradientStringArray,
        gradientTabs
      );
      addPropertyDocString( "backgroundImage",
        "string",
        gradientDocStringArr,
        gradientTabs - 1
      );
      addKeyValueToFileStr( "backgroundImage",
        gradient[ "backgroundImage" ],
        gradientStringArray,
        gradientTabs
      );
    },
    tabs,
    true,
    name
  );
  fileStrArr.push( " " );
  gradientString.forEach( item => {
    fileStrArr.push( item );
  } );
};

/**
 * @param {string} name
 * @param {genObjectCallBack} cb
 * @param {number} tabs
 * @param {boolean} insideObject
 * @param objectName
 * @return {string[]} docAndFileString
 * @return {string} objectName
 */
const generateObjectStartAndClose = ( name, cb, tabs = 0, insideObject = false,
  objectName = "" ) => {
  const fileStrArr = [];
  const docStringArray = [];
  let str = "";
  if( insideObject ){
    
    str += `${ name }: {`;
  }else{
    str += `export const ${ name } = {`;
  }
  
  generateDocStringStart( name, docStringArray, tabs );
  
  addStrToArrayStr( str, fileStrArr, tabs );
  cb( name, fileStrArr, docStringArray, tabs + 1 );
  str = "}";
  if( insideObject ){
    str += ",";
  }
  
  addStrToArrayStr( str, fileStrArr, tabs );
  closeOutDocStringArr( docStringArray, tabs );
  
  return [ ...docStringArray, ...fileStrArr ];
};
/**
 * @typedef {Function} genObjectCallBack
 * @param {string} name
 * @param {string[]} fileStrArr
 * @param {string[]} docStringArray
 * @param {number} tabs
 */

/**
 *
 */
const closeOutDocStringArr = ( docStringArr, tabs ) => {
  addStrToArrayStr( " */", docStringArr, tabs );
};

/**
 * @param {string} name
 * @param {string[]} docString
 * @param {number} [tabs]
 */
const generateDocStringStart = ( name, docString, tabs = 0 ) => {
  addStrToArrayStr( `/**`, docString, tabs );
  addStrToArrayStr( ` * @typedef {Object} ${ pascalCase( name ) }`,
    docString,
    tabs
  );
};

/**
 * @param {string} key
 * @param {string} color
 * @param {string[]} fileStrArr
 * @param {string[]} docStringArray
 * @param {number = 0} [tabs]
 * @param {number} [numberOfItems]
 * @param {number} [percentage]
 *
 */
const generateDarker = ( key, color, fileStrArr, docStringArray, tabs = 0,
  numberOfItems = 5, percentage = .1, ) => {
  for( let i = numberOfItems; i > 0; i-- ){
    let newKey = `${ key }_DARKER_${ i }`;
    addKeyValueToFileStr( newKey,
      darken( percentage * i, color ),
      fileStrArr,
      tabs
    );
    addPropertyDocString( newKey, "Color", docStringArray );
  }
  
};

/**
 * @param {string} key
 * @param {string} color
 * @param {string[]} fileStrArr
 * @param {string[]} docStringArray
 * @param {number = 0} [tabs]
 * @param {number} [numberOfItems]
 * @param {number} [percentage]
 *
 */
const generateLighter = ( key, color, fileStrArr, docStringArray, tabs = 0,
  numberOfItems = 5, percentage = .1, ) => {
  for( let i = 0; i < numberOfItems; i++ ){
    let newKey = `${ key }_LIGHTER_${ i + 1 }`;
    addKeyValueToFileStr( newKey,
      lighten( percentage * ( i + 1 ), color ),
      fileStrArr,
      tabs
    );
    addPropertyDocString( newKey, "Color", docStringArray );
  }
};

/**
 * @param {string} key
 * @param {string} value
 * @param {string[]} fileStrArr
 * @param {number} tabs
 */
const addKeyValueToFileStr = ( key, value, fileStrArr, tabs = 0 ) => {
  addStrToArrayStr( `${ key }: "${ value }",`, fileStrArr, tabs );
};

/**
 * @param {string} key
 * @param {string} propType
 * @param {string[]} docString
 * @param {number} tabs
 */
const addPropertyDocString = ( key, propType, docString, tabs = 0 ) => {
  const string = ` * @property {${ propType }} ${ key } `;
  addStrToArrayStr( string, docString, tabs );
};

/**
 * @param {string} str
 * @param {string[]} stringArray
 * @param {number} tabs
 */
const addStrToArrayStr = ( str, stringArray, tabs = 0 ) => {
  let strValue = "";
  while( tabs !== 0 ){
    strValue += `\t`;
    tabs--;
  }
  strValue += `${ str }`;
  stringArray.push( strValue );
};

const colors = {
  PRIMARY_COLOR: "#36405C", SECONDARY_COLOR: "#4CB69F"
};

generateColorPallet( colors );
