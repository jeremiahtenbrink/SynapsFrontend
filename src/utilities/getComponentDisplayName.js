export const getComponentDisplayName = ( Component ) => Component.displayName ||
  Component.name || "Component";
