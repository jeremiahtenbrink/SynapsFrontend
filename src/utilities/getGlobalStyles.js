import { createGlobalStyle } from "styled-components";
import { normalize } from "styled-normalize";

const GlobalStyle = createGlobalStyle`
${ normalize }
  body {
        font-family: "Source Sans Pro";
  }  
`;

export const getGlobalStyles = () => GlobalStyle;
