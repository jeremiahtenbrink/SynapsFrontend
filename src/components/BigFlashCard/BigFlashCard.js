import React, { useState } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { darken } from "polished";
import TopCard from "../../svgComponents/TopCard.js";
import { APP_VIEW_DESKTOP } from "../../utilities/constants.js";
import BottomQuizCards from "../../svgComponents/BottomQuizCards.js";

/**
 * Big Flash Card
 *
 * @component
 * @example
 * const flashCard = {question: "Question?", answer: "Answer"}
 * return (
 *  <BigFlashCard flashCard={flashCard} />
 *  )
 */
const BigFlashCard = ( { flashCard, appView } ) => {
  const [ position, setPosition ] = useState( "front" );
  
  const flipCard = () => {
    
    const newPos = position === "front" ? "back" : "front";
    setPosition( newPos );
  };
  console.log( flashCard.question );
  return ( <StyledCardContainer data-testid="StyledCardContainer"
                                style={ { position: "relative" } }
                                onClick={ flipCard }>
      
      <TopCard position={ position } question={ flashCard.question }
               answer={ flashCard.answer } front_url={ flashCard.image_front }
               back_url={ flashCard.image_back }/>
      <BottomQuizCards/>
    </StyledCardContainer>
  
  );
};

BigFlashCard.prototypes = {
  flashCard: PropTypes.objectOf( {
    question: PropTypes.string.isRequired, answer: PropTypes.string.isRequired,
  } ).isRequired,
};

const StyledCardContainer = styled.div`
margin-top: 2rem;
  height: 390px;
  width: 283px;
  position: relative;
`;

const StyledCard = styled.div`
  && {
    margin: 0px auto 0 auto;
    z-index: 2;
    background: ${ props => props.position === "front" ? "#FFFBF4" :
  darken( .1, "#4ea699" ) };
    color: ${ props => ( props.position === "front" ? "#1b1414c9" : "white" ) };
    width: ${ props => ( props.theme.appView === APP_VIEW_DESKTOP ? "352PX" :
  "285PX" ) };
    height: ${ props => ( props.theme.appView === APP_VIEW_DESKTOP ? "460PX" :
  "421PX" ) };
    border-radius: 11px;
    > .ant-card-body {
      min-height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }
   
  }
  transition: all 1s;
`;
const CardText = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  font-size: 20px;
  font-weight: 900;
  transition: all 1s;
`;
export default BigFlashCard;
