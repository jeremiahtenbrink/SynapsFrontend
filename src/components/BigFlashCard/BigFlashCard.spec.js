import React from "react";
import { customRender, fireEvent } from "../../utilities/test-utils";
import BigFlashCard from "./BigFlashCard";

/**
 * Test the big flash card
 */

describe( "Big Flash Card", () => {
  
  test( "BigFlashCard render correctly", () => {
    const { container } = customRender( <BigFlashCard flashCard={ {
      question: "This is" +
        ' a question',
    }}/>);
    expect(container).toMatchSnapshot();
  });
  
  test('position front', () => {
    const {container, getByText} = customRender(<BigFlashCard
      flashCard={{question: 'question', answer: 'answer'}}/>);
    expect(getByText('question')).toBeInTheDocument();
  });
  test('position back', () => {
    const {container, getByTestId, queryByText, debug} = customRender(
      <BigFlashCard
        flashCard={{question: 'question', answer: 'answer'}}/>);
    const card = getByTestId('StyledCardContainer');
    fireEvent.click(card);
    expect(queryByText('answer')).toBeInTheDocument();
    //debug(queryByText('answer'))
  });
  
});