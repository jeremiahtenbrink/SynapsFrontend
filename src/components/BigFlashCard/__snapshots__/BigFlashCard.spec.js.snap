// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big Flash Card BigFlashCard render correctly 1`] = `
<div>
   
   
</div>
`;
